server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/configuration/CassandraMailQueueViewConfiguration.java

```diff
@@ -44,72 +44,88 @@
 
         @FunctionalInterface
         interface RequireSliceWindow {
-            ReadyToBuild sliceWindow(Duration sliceWindow);
+            RequireSizeMetricsEnabled sliceWindow(Duration sliceWindow);
+        }
+
+        @FunctionalInterface
+        interface RequireSizeMetricsEnabled {
+            ReadyToBuild sizeMetricsEnabled(boolean sizeMetricsEnabled);
```

Comment:
This configuration parameter should be optional, hence I don't believe staged builder is the best way to go.